lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -4,8 +4,18 @@ defmodule Kernel.WarningTest do
   use ExUnit.Case
   import ExUnit.CaptureIO
 
-  defp capture_err(fun) do
-    capture_io(:stderr, fun)
+  defp capture_err(fun) when is_function(fun), do: capture_io(:stderr, fun)
+
+  defp capture_err(source, op \\ :eval) when op in [:eval, :quote, :compile] do
```

Comment:
Maybe we introduce functions called `capture_eval`, `capture_quoted`, and `capture_compile` instead of the additional argument? It will be a bit cleaner to type. :)